CharlieBackend.Api/Controllers/MentorsController.cs

```diff
@@ -63,15 +63,18 @@ public async Task<ActionResult<List<MentorModel>>> GetAllMentors()
         [HttpPut("{id}")]
         public async Task<ActionResult> PutMentor(long id, UpdateMentorModel mentorModel)
         {
-            if (!ModelState.IsValid)
+
+            var mentor = await _mentorService.GetMentorByIdAsync(id);
+
+            if (mentor == null)
             {
-                return BadRequest();
+                return BadRequest("Can not find mentor!");
             }
 
-            var isEmailChangableTo = await _accountService
-                    .IsEmailChangableToAsync(mentorModel.Email);
+            var isEmailChangable = await _accountService
+                .IsEmailChangableToAsync(mentor.AccountId, mentorModel.Email);
 
-            if (!isEmailChangableTo)
+            if (!isEmailChangable)
             {
                 return StatusCode(409, "Email is already taken!");
             }
```

Comment:
Please move this validation inside MendorService

Reply:
I did not touch anything in MentorsController.cs as  SnipeFrost (Eugene) is rewriting it. Target was to save workable state of existing model until new implementation is done (to avoid non-workings APIs).